app/scripts/modules/core/src/pipeline/config/triggers/artifacts/s3/defaultS3.artifact.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { module } from 'angular';
+
+import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
+import { IArtifact } from 'core/domain/IArtifact';
+import { PipelineConfigProvider } from 'core/pipeline';
+import { isNil } from 'lodash';
+
+export const DEFAULT_S3_ARTIFACT = 'spinnaker.core.pipeline.trigger.artifact.defaultS3';
+module(DEFAULT_S3_ARTIFACT, [
+  PIPELINE_CONFIG_PROVIDER,
+]).config((pipelineConfigProvider: PipelineConfigProvider) => {
+  pipelineConfigProvider.registerArtifactKind({
+    label: 'S3',
+    description: 'A S3 object.',
+    key: 'default.s3',
+    isDefault: true,
+    isMatch: false,
+    controller(artifact: IArtifact) {
+      'ngInject';
+      this.artifact = artifact;
+      this.artifact.type = 's3/object';
+
+      this.onReferenceChange = () => {
+        const ref = this.artifact.reference;
+        if (isNil(ref)) {
+          return;
+        }
+
+        if (ref.indexOf('#') >= 0) {
+          const split = ref.split('#');
+          this.artifact.name = split[0];
+          this.artifact.version = split[1];
+        } else {
+          this.artifact.name = ref;
+        }
+      };
+    },
+    controllerAs: 'ctrl',
+    template: `
+<div class="col-md-12">
+  <div class="form-group row">
+    <label class="col-md-2 sm-label-right">
+      Object path
+      <help-field key="pipeline.config.expectedArtifact.defaultS3.reference"></help-field>
```

Comment:
this key and the one in s3.artifact.ts need correlating help text entries in `app/scripts/modules/core/src/help/help.contents.ts`

Reply:
Awesome, I'll work that out. Should have figured that was referencing _something_ ;)